gdc_client/upload/client.py

```diff
@@ -510,73 +527,78 @@ def upload_parts(self):
             manager = Manager()
             self.ns = manager.Namespace()
             self.ns.completed = 0
+
         part_amount = int(math.ceil(self.file_size / float(self.upload_part_size)))
+
         self.total_parts = part_amount
-        self.pbar = ProgressBar(
-            widgets=[Percentage(), Bar()], maxval=self.total_parts
-        ).start()
-        try:
-            for i in range(part_amount):
-                offset = i * self.upload_part_size
-                num_bytes = min(self.file_size - offset, self.upload_part_size)
-                if not self.multiparts.uploaded(i + 1):
-                    args_list.append(
-                        [
-                            self.file_path,
-                            offset,
-                            num_bytes,
-                            self.url,
-                            self.upload_id,
-                            i + 1,
-                            self.headers,
-                            self.verify,
-                            self.pbar,
-                            self.ns,
-                        ]
-                    )
-                else:
-                    self.total_parts -= 1
-            if self.total_parts == 0:
-                return
-            self.pbar.maxval = self.total_parts
-
-            pool = Pool(processes=self.processes)
-            pool.map_async(upload_multipart_wrapper, args_list).get(9999999)
-            pool.close()
-            pool.join()
-        except KeyboardInterrupt:
-            log.error("Caught KeyboardInterrupt, terminating workers")
-            pool.terminate()
-            pool.join()
-            raise Exception("Process canceled by user")
+        for i in range(part_amount):
+            offset = i * self.upload_part_size
+            num_bytes = min(self.file_size - offset, self.upload_part_size)
+            if not self.multiparts.uploaded(i + 1):
+                args_list.append(
+                    [
+                        self.file_path,
+                        offset,
+                        num_bytes,
+                        self.url,
+                        self.upload_id,
+                        i + 1,
+                        self.headers,
+                        self.verify,
+                        self.ns,
+                        self.debug,
+                    ]
+                )
+            else:
+                self.total_parts -= 1
+
+        if self.total_parts == 0:
+            return
+
+        pbar = tqdm(
```

Comment:
might be a good place to use `with` so you don't have to manually close the pbar

https://pypi.org/project/tqdm/#examples-and-advanced-usage